src/test/scala/outwatch/LifecycleHookSpec.scala

```diff
@@ -132,20 +132,15 @@ class LifecycleHookSpec extends JSDomAsyncSpec {
       node <- div(message, onSnabbdomUpdate --> sink1)(onSnabbdomUpdate --> sink2)
     } yield node
 
-    val res1 = for {
-      _ <- renderUnsafe(node)
-    } yield {
-      switch1 shouldBe false
-      switch2 shouldBe false
-    }
+    renderUnsafe(node).value.get.get
```

Comment:
It is needed to await the future, using this imperative style. Using flatmap somehow does not work with `onNext`

Reply:
I understand now, that this would throw if the future was not executed successfully or not executed at all. But why doesn't it work in the for-loop?

Reply:
Please try it out if you have the time, on the phone now.

Reply:
Some kind of race condition, I guess. It is the equivalent of unsafeRunSync

Reply:
> It is needed to await the future
`.value` actually does not await. It just takes the current value...
(see https://www.scala-lang.org/api/current/scala/concurrent/Future.html#value:Option[scala.util.Try[T]])

I'm investigating in the for-loop problem now.

Reply:
Ok thanks

Reply:
It looks like we cannot rely on the backpressure of `message.onNext` to give us a `Continue` *after* `onSnabbdomUpdate` was called and its emitter has written into the sink. There seems to be some asyncronicity in there somewhere. I suspect it to be the Emitter. This is especially strange since we have a synchronous scheduler...

Still, thank you very much for taking the time and make this PR.

Maybe @cornerman has an idea why we cannot rely on the backpressure?

Reply:
@fdietze I suspected it was something around `onNext`

Reply:
@fdietze Alexandru from Monix says nothing has changed wrt. backpressure/scheduler behavior between RC1 and RC2.